src/lapack/backends/cusolver/cusolver_batch.cpp

```diff
@@ -79,251 +100,645 @@ void getri_batch(sycl::queue &queue, std::int64_t n, sycl::buffer<std::complex<d
+                     std::int64_t nrhs, sycl::buffer<TYPE> &a, std::int64_t lda,                  \
+                     std::int64_t stride_a, sycl::buffer<std::int64_t> &ipiv,                     \
+                     std::int64_t stride_ipiv, sycl::buffer<TYPE> &b, std::int64_t ldb,           \
+                     std::int64_t stride_b, std::int64_t batch_size,                              \
+                     sycl::buffer<TYPE> &scratchpad, std::int64_t scratchpad_size) {              \
+        return getrs_batch(#CUSOLVER_ROUTINE, CUSOLVER_ROUTINE, queue, trans, n, nrhs, a, lda,    \
+                           stride_a, ipiv, stride_ipiv, b, ldb, stride_b, batch_size, scratchpad, \
+                           scratchpad_size);                                                      \
+    }
+
+GETRS_STRIDED_BATCH_LAUNCHER(float, cusolverDnSgetrs)
+GETRS_STRIDED_BATCH_LAUNCHER(double, cusolverDnDgetrs)
+GETRS_STRIDED_BATCH_LAUNCHER(std::complex<float>, cusolverDnCgetrs)
+GETRS_STRIDED_BATCH_LAUNCHER(std::complex<double>, cusolverDnZgetrs)
+
+#undef GETRS_STRIDED_BATCH_LAUNCHER
+
+template <typename Func, typename T>
+inline void getrf_batch(const char *func_name, Func func, sycl::queue &queue, std::int64_t m,
+                        std::int64_t n, sycl::buffer<T> &a, std::int64_t lda, std::int64_t stride_a,
+                        sycl::buffer<std::int64_t> &ipiv, std::int64_t stride_ipiv,
+                        std::int64_t batch_size, sycl::buffer<T> &scratchpad,
+                        std::int64_t scratchpad_size) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+
+    overflow_check(m, n, lda, stride_a, stride_ipiv, batch_size, scratchpad_size);
+
+    // cuSolver legacy api does not accept 64-bit ints.
+    // To get around the limitation.
+    // Create new buffer with 32-bit ints then copy over results
+    std::uint64_t ipiv_size = stride_ipiv * batch_size;
+    sycl::buffer<int, 1> ipiv32(sycl::range<1>{ ipiv_size });
+    sycl::buffer<int> devInfo{ batch_size };
+
+    queue.submit([&](sycl::handler &cgh) {
+        auto a_acc = a.template get_access<sycl::access::mode::read_write>(cgh);
+        auto ipiv32_acc = ipiv32.template get_access<sycl::access::mode::write>(cgh);
+        auto devInfo_acc = devInfo.template get_access<sycl::access::mode::write>(cgh);
+        auto scratch_acc = scratchpad.template get_access<sycl::access::mode::write>(cgh);
+        onemkl_cusolver_host_task(cgh, queue, [=](CusolverScopedContextHandler &sc) {
+            auto handle = sc.get_handle(queue);
+            auto a_ = sc.get_mem<cuDataType *>(a_acc);
+            auto ipiv_ = sc.get_mem<int *>(ipiv32_acc);
+            auto devInfo_ = sc.get_mem<int *>(devInfo_acc);
+            auto scratch_ = sc.get_mem<cuDataType *>(scratch_acc);
+            cusolverStatus_t err;
+
+            // Uses scratch so sync between each cuSolver call
+            for (int64_t i = 0; i < batch_size; ++i) {
```

Comment:
I don't know how strict it is, but int64_t here is used without std namespace, a bit different vs everywhere else, and 32 bit int type is used not as int32_t.

Reply:
I have added the `std` namespace. 

> and 32 bit int type is used not as int32_t.

Could you clarify what you are referring to, which 32 bit int type is not used as a int32_t and in what way?

Reply:
The comment regarding legacy API makes use of 32-bit int a special case vs use of int64_t, so this special case could be explicitly highlighted with explicit fixed width type int32_t, not implementation dependent int type. In the current building environment int equals to int32_t. Use of int32_t would be just for code readability (highlight special case) and consistency (use only fixed-width integer types). This is just for your consideration.

Reply:
I agree it would be better to have this as explicitly stated as int32_t rather than int. I think it should be a separate PR as the non-batched implementation also uses `int` in this way so it would be out of scope of this PR.